script.js

```diff
@@ -1,4 +1,230 @@
+var stage2 = function (chosenCndtn) {
+  var resultMessage;
+  // Input validation: if user does not enter a valid win condition
+  if (chosenCndtn != HIGHEST && chosenCndtn != LOWEST) {
+    resultMessage = `You have entered an invalid input.<br><br>Please enter "${HIGHEST}" to have the the win condition be ${"highest number wins".italics()}, or enter "${LOWEST}" to have the win condition be ${"lowest number wins".italics()}.`;
+    return resultMessage;
+  }
+  // Valid input, so we update winCndtn based on input
+  winCndtn = chosenCndtn;
+  resultMessage = `You have selected the win condition as ${(winCndtn + " number wins").italics()}.<br><br>
+  Now enter "${ON}" to have Knockout Mode turned on, or enter "${OFF}" to have Knockout Mode turned off.`;
+  gameMode = CHOOSE_KNOCKOUT; // Update game mode to next stage
+  return resultMessage;
+};
+
+// Third stage of game: Choose Knockout Mode setting
+var stage3 = function (onOrOff) {
+  var resultMessage;
+  // Input validation: if user does not enter "on" or "off"
+  if (onOrOff != ON && onOrOff != OFF) {
+    resultMessage = `You have entered an invalid input.<br><br>Please enter "${ON}" to have Knockout Mode turned on, or enter "${OFF}" to have Knockout Mode turned off.`;
+    return resultMessage;
+  }
+  // Valid input, so we update koSetting based on input
+  koSetting = onOrOff;
+  gameMode = CHOOSE_NUM_DICE; // Update game mode to next stage
+  resultMessage = `You have selected Knockout Mode to be turned ${koSetting}.<br><br>These are the settings that you have chosen:<br>Number of players: ${numPlayers}<br>Win condition: ${winCndtn} number wins<br>Knockout Mode: ${koSetting}<br><br>You are now ready to start the first round of Beat That!<br>Enter the number of dice that will be rolled for each player in this round.`;
+  return resultMessage;
+};
+
+// Choose number of dice to be rolled for the round and then play the round
+var stage4 = function (numDiceInput) {
+  var resultMessage;
+  // Input validation: need at least 1 die to be rolled. Limit number of dice to 8 for performance reasons.
+  if (!(Number(numDiceInput) >= 1 && Number(numDiceInput) <= 8)) {
+    resultMessage = `You have entered an invalid input.<br>Please enter a number between 1 and 8. All decimals will be rounded up.`;
+    return resultMessage;
+  }
+  // Valid input, so we update numDice based on input rounded up
+  numDice = Math.ceil(Number(numDiceInput));
+  // Play the round
+  if (koSetting == ON) {
+    return playRoundKOMode(numDice);
+  } else {
+    return playRound(numDice);
+  }
+};
+
+// Play the round when Knockout Mode is turned off
+var playRound = function (numDice) {
+  var resultMsg = "";
+  // Roll dice and calculate score for each player
+  for (var p = 0; p < players.length; p++) {
+    var diceRolls = [];
+    for (var i = 0; i < numDice; i++) {
+      diceRolls.push(rollDice());
+    }
+    var roundScore = calcNumber(diceRolls);
+    players[p].playerScore += roundScore;
```

Comment:
This group of code also seems to be repeated. A good sign that you can create a helper function for this